### PR TITLE
Retry Custom Confirmation

### PR DESCRIPTION
**Pull request by ericjordanmossman**
Add support for retrying a [custom confirmation](https://docs.mongodb.com/realm/authentication/email-password/#run-a-confirmation-function) function.

### CODE REVIEW

**Review by jsflax — COMMENTED** [review]
This should be tested from Swift and also needs a Combine version.

**Comment by ericjordanmossman** [comment]
Swift testing occurs [here](https://github.com/realm/realm-cocoa/blob/c72fc35afdef8f3bcb4be64eb57ea361041b3696/Realm/ObjectServerTests/SwiftObjectServerTests.swift#L903), as part of the emailPasswordAuth unit.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 x.y.z Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
-* None.
+* Add support for retrying a [custom confirmation](https://docs.mongodb.com/realm/authentication/email-password/#run-a-confirmation-function) function.
+- `App.emailPasswordAuth.retryCustomConfirmation(email:completion:)`
+- `[App.emailPasswordAuth retryCustomConfirmation:completion:]`
```
Comment (jsflax) [inline CHANGELOG.md:6]: Can collapse this into a single line.

**Review by jsflax — APPROVED** [review]

**Comment by ericjordanmossman** [comment]
Merging with known / suspected unrelated errors.
[Jenkins: swiftpm-debug 13.0](https://ci.realm.io/job/objc_pr/target=swiftpm-debug,xcode_version=13.0/8165/consoleFull)
<details>
Jun 22 12:39:30 Test Case '-[ArrayPropertyTests testAllMethodsCheckForInvalidation]' started.
Jun 22 12:39:30 Exited with signal code 10
</details>

[Jenkins: swiftui-ios 13.0](https://ci.realm.io/job/objc_pr/target=swiftui-ios,xcode_version=13.0/8165/consoleFull)

<details>
Jun 22 12:36:53 Failing tests:
Jun 22 12:36:53 SwiftUITestHostUITests:
Jun 22 12:36:53 SwiftUITests.testSampleApp()
Jun 22 12:36:53 SwiftUITests.testUnmanagedObjectState()
</details>

[Jenkins: osx 12.5.1](https://ci.realm.io/job/objc_pr/target=osx,xcode_version=12.5.1/8165/console)
&
[Jenkins: cocoapods-osx 12.5.1](https://ci.realm.io/job/objc_pr/target=cocoapods-osx,xcode_version=12.5.1/8165/console)

<details>
No Xcode found with version 12.5.1
</details>

**Pull request merged**